Add pool suggestions displaying frequently used groups

norbye opened the pull request:
# Description

Added some suggestions of frequently used group combinations to make life simpler for event organizers.

The group combinations disable if more comprehensive groups have been selected, ensuring the the minimum amount of groups required is stored, while maintaining the group overlap and so on.

The functionality does not at all affect the input, so the users can still use it as they used to, but I believe this could resolve the errors of adding the wrong groups as this is simpler than typing them in.

The suggestions listen to the content of the list, so even if it is not used to select the groups, it can be used as an indicator of what groups you've already selected.

# Result

If you've made visual changes, please check the boxes below and include images showing the changes. Descriptions are appreciated.

- [x] Changes look good on both light and dark theme.
- [x] Changes look good with different viewports (mobile, tablet, etc.).
- [x] Changes look good with slower Internet connections.

## Preview

https://github.com/webkom/lego-webapp/assets/13599770/e68f0d6c-6af9-44f1-8782-8b5daed58dec

# Testing

- [ ] I have thoroughly tested my changes.

Please describe what and how the changes have been tested, and provide instructions to reproduce if necessary.

---

Resolves ABA-848

eikhr commented on app/routes/events/components/EventEditor/renderPools.tsx:
```diff
@@ -15,11 +16,6 @@ type poolProps = {
   eventStatusType: EventStatusType;
 };
 
-const highWarning = (value) =>
-  value && value >= 500
-    ? 'Åpent arrangement gir uendelig plasser 😁'
-    : undefined;
-
```
Why remove this?

ivarnakken replied:
Warning are currently not supported with react-final-form

norbye replied:
Yeah, I figure we might re-add warnings where we want to at some point, but for now it's just dead code throwing console errors - so I'd rather remove it

eikhr commented on app/routes/events/components/EventEditor/PoolSuggestions.tsx:
```diff
@@ -0,0 +1,151 @@
+  },
+];
+
+type PoolSuggestionType = React.ComponentProps<
+  typeof SelectInput<{ label: string; value: number }>
+>['SuggestionComponent'];
```
Could we maybe just export the `SuggestionComponent`-type from `SelectInput.tsx`?

norbye replied:
Excellent idea, updated now

eikhr reviewed (approved):
Looks great! This makes it a lot easier to select the typical event groups, and I think it will really help people to understand the implicit groups too.

itsisak commented on app/routes/events/components/EventEditor/EventEditor.css:
```diff
@@ -164,3 +172,31 @@
+      cursor: not-allowed;
+      background-color: var(--color-green-3);
+
+      @media (prefers-color-scheme: dark) {
```
```suggestion
      [data-theme='dark'] {
```
`prefers-color-scheme: dark` is still true in light-mode when system/browser prefers dark

norbye replied:
Sweet, that explains why it wasn't working(: ended up removing it, but will add the data-theme stuff if the styling should require it

itsisak commented on app/routes/events/components/EventEditor/EventEditor.css:
```diff
@@ -164,3 +172,31 @@
+  }
+}
+
+@media (prefers-color-scheme: dark) {
```
```suggestion
[data-theme='dark'] {
```

itsisak reviewed (approved):
Great idea!

ivarnakken reviewed (approved):
Very useful feature!

I am, however, not sure about the use of green. We don't really use that elsewhere to indicate an "active" state. Might I suggest the the `var(--lego-font-color)` color, like we utilize for navigation tabs and buttons?

norbye commented:
> I am, however, not sure about the use of green. We don't really use that elsewhere to indicate an "active" state. Might I suggest the the `var(--lego-font-color)` color, like we utilize for navigation tabs and buttons?

Absolutely, changed to use lego-font-color now and updated the preview recording
